rfcs/0001-image-trait.md

```diff
@@ -0,0 +1,127 @@
+
+* RFC PR: [rust-cv/rfcs#N](https://github.com/rust-cv/rfcs/pull/N)
+* Tracking Issue [rust-cv/rfcs#N](https://github.com/rust-cv/rfcs/issues/N)
+
+# Summary
+
+An `Image` trait to allow interoperability of image data.
+
+# Motivation
+
+Multiple existing crates implement concrete types that hold image data either
+explicitly or implicitly. If each of these implemented a single, fundamental
+trait that allowed specification of how to access image data across
+implementations, image operations could be written generically that could
+operate on any of the data types. This would enable, for example, image
+processing routines to be written which would work equally well regardless of
+how the data types is defined.
+
+# Prior Art
+
+## [`machine_vision_formats::ImageStride` v0.1.1](https://docs.rs/machine-vision-formats/0.1.1/machine_vision_formats/trait.ImageStride.html)
+
+Here we consider
+[`machine_vision_formats::ImageStride`](https://docs.rs/machine-vision-formats/0.1.1/machine_vision_formats/trait.ImageStride.html)
+together with its supertraits
+[`machine_vision_formats::ImageData`](https://docs.rs/machine-vision-formats/0.1.1/machine_vision_formats/trait.ImageData.html)
+and
+[`machine_vision_formats::Stride`](https://docs.rs/machine-vision-formats/0.1.1/machine_vision_formats/trait.Stride.html).
+
+### What's good
+
+A simple mostly-trait only. Provides a "view" of existing structure and enum
+definitions, allowing low-cost consumption of image data from a variety of
+sources. This trait be implemented by crates which provide their own data types
+that could be used to hold image data.
+
+### What's not so good
```

Comment:
So one thing is there could (probably should) be convenience functions to access pixels at given x,y coordinates (or specific channel values (x,y,z). And something to allow for moving for moving windows such as https://docs.rs/ndarray/latest/ndarray/struct.ArrayBase.html#method.windows 

These and the issues already mentioned in the "what's not so good" are the only things that would stop me from moving over the ndarray-vision algorithms to be implemented in terms of `ImageStride`